apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/TableStatsCollectorUtil.java

```diff
@@ -127,14 +126,11 @@ public static IcebergTableStats populateStorageStats(
     long sumOfTotalDirectorySizeInBytes = 0;
     long numOfObjectsInDirectory = 0;
     try {
-      RemoteIterator<LocatedFileStatus> it = fs.listFiles(new Path(table.location()), true);
-      while (it.hasNext()) {
-        LocatedFileStatus status = it.next();
-        numOfObjectsInDirectory++;
-        sumOfTotalDirectorySizeInBytes += status.getLen();
-      }
+      ContentSummary contentSummary = fs.getContentSummary(new Path(table.location()));
```

Comment:
getContentSummary can cause namenode issues. It obtains a lock throughout the process of this computation and this will have an impact on namenode performance for larger directories. It is as expensive as ls but atleast ls drops the lock in between some batching it does internally but getContentSummary will not. This will cause a performance hit and also could be throttled since cost assigned to this could be higher while serving this request.